client/components/App.jsx

```diff
@@ -75,9 +75,17 @@ class App extends React.Component {
   }
 
   addToOutfit() {
-    let outfitArray = this.state.yourOutfit;
-    outfitArray.push(this.state.product);
-    this.setState({yourOutfit: outfitArray});
+    let productAlreadyInOutfit = false;
+    let outfitArray = this.state.yourOutfit.slice();
+    for (let x = 0; x < outfitArray.length; x++) {
+      if (outfitArray[x].id === this.state.product.id) {
+        productAlreadyInOutfit = true;
```

Comment:
Let me preface this by saying that it's the end of the day, so I might not be at my sharpest, BUT - 'this.state.product.id' is just the product were currently viewing on the page, right? It hasn't, necessarily, been added to 'your outfit'. so what is line 81 testing?

Reply:
@zrealberg No, definitely a good question as I had to ponder over this a lot. What I'm trying to do is check to see if the product that we're currently viewing on the page has already been added to the outfit or not. If it has, then this method will not add it again. I couldn't think of way to run this check in constant time. Since there's no way to know where in the outfit array the product is, I wrote a for loop to check each item in the outfit array. Is this a redundant check?